.github/ISSUE_TEMPLATE/testplan.md

```diff
@@ -235,6 +235,24 @@ Minikube is the only caveat - it's not reachable publicly so don't run a proxy t
   * [ ] Verify that clicking on a rows connect button renders a dialogue on manual instructions with `Step 2` login value matching the rows `name` column
   * [ ] Verify searching for `name` or `labels` in the search bar works
   * [ ] Verify you can sort by `name` colum
+* [ ] Test Kubernetes auto-discovery:
+  * [ ] Verify that Azure AKS clusters are discovered and enrolled for different Azure Auth configs:
+    * [ ] Local Accounts only
+    * [ ] Azure AD
+    * [ ] Azure RBAC
+  * [ ] Verify that AWS EKS clusters are discovered and enrolled
+* [ ] Kubernetes Secret storage for Agent's Identity
+    * [ ] Install Teleport 11 agent with a short-lived token  
+      * [ ] Validate if the Teleport is installed as a Kubernetes `Statefulset`
+      * [ ] Restart the agent after token TTL expires to see if it reuses the same identity.
+    * [ ] Upgrade from Teleport 10 agent with storage
+      * [ ] Validate if agent identity was read from storage and stored in the secret without generating a new one
+      * [ ] Validate if the Teleport is still running as a `Statefulset` resource and if it contains the new ENV variables
+    * [ ] Upgrade from Teleport 10 agent without storage
+      * [ ] Validate if the agent identity is created and stored in the secret using the long-lived token.
+      * [ ] Validate if the Teleport Kubernetes `Deployment` was correctly converted into a Statefulset and if the old `Deployment` object was removed after a successful upgrade
+    * [ ] Force cluster CA rotation
```

Comment:
Since this is a one-off (only applies to 10 -> 11 upgrade), how about we don't add it to the testplan.md and just add it to Teleport 11 specific test plan ticket? https://github.com/gravitational/teleport/issues/16951